File: app/backend/java/payment-service/src/main/java/com/srbms/paymentservice/dto/FareCalculationRequestDTO.java
package com.srbms.paymentservice.dto;

import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotNull;
import java.math.BigDecimal;

public class FareCalculationRequestDTO {

    @NotNull(message = "Distance in km is required")
    @DecimalMin(value = "0.1", message = "Distance must be greater than zero")
    private BigDecimal distanceKm;

    private Integer vehicleTypeId;
    private String vehicleTypeName; // Sedan, SUV, Bike, Auto
    private BigDecimal surgeMultiplier = BigDecimal.ONE;

    public FareCalculationRequestDTO() {}

    public BigDecimal getDistanceKm() {
        return distanceKm;
    }

    public void setDistanceKm(BigDecimal distanceKm) {
        this.distanceKm = distanceKm;
    }

    public Integer getVehicleTypeId() {
        return vehicleTypeId;
    }

    public void setVehicleTypeId(Integer vehicleTypeId) {
        this.vehicleTypeId = vehicleTypeId;
    }

    public String getVehicleTypeName() {
        return vehicleTypeName;
    }

    public void setVehicleTypeName(String vehicleTypeName) {
        this.vehicleTypeName = vehicleTypeName;
    }

    public BigDecimal getSurgeMultiplier() {
        return surgeMultiplier;
    }

    public void setSurgeMultiplier(BigDecimal surgeMultiplier) {
        this.surgeMultiplier = surgeMultiplier;
    }
}

File: app/backend/.net/SmartRide.PaymentService/Models/DTOs/FareCalculationRequestDTO.cs
using System.ComponentModel.DataAnnotations;

namespace SmartRide.PaymentService.Models.DTOs;

public class FareCalculationRequestDTO
{
    [Required(ErrorMessage = "Distance in km is required")]
    [Range(typeof(decimal), "0.1", "79228162514264337593543950335", ErrorMessage = "Distance must be greater than zero")]
    public decimal DistanceKm { get; set; }

    public int? VehicleTypeId { get; set; }
    public string? VehicleTypeName { get; set; }
    public decimal SurgeMultiplier { get; set; } = 1.0m;
}
